plot/plot.app.js

```diff
@@ -683,14 +695,56 @@
+                                                                    //Example : traceLabel: <a href="/chaise/recordset/#2/Gene_Expression:Specimen/*::facets::N4IghgdgJiBcDaoDOB7ArgJwMYFM6JHQBcAjdafEAQQiIEsyoBPEAGmtoZWYH0AVHEiJIeAYRQRUGemgC2PAGYBrHCwC6AX3YAlAJIAREGvZYAFijq4klAIoBaAMwBpAKwA2I1uTpseBKGIyNAoEDnpGFnYacO4mfkFhMUwwegkeAGUiFLQRZVVPHQMjE3NLQVs7AE03AEYAWU9NIA&pcid=plotApp&ppid=2oio1qej2i6m2bv62ox21h1q" target="_blank">Imaging: Histology</a>
+                                                                    // extractedLink = /chaise/recordset/#2/Gene_Expression:Specimen/*::facets::N4IghgdgJiBcDaoDOB7ArgJwMYFM6JHQBcAjdafEAQQiIEsyoBPEAGmtoZWYH0AVHEiJIeAYRQRUGemgC2PAGYBrHCwC6AX3YAlAJIAREGvZYAFijq4klAIoBaAMwBpAKwA2I1uTpseBKGIyNAoEDnpGFnYacO4mfkFhMUwwegkeAGUiFLQRZVVPHQMjE3NLQVs7AE03AEYAWU9NIA&pcid=plotApp&ppid=2oio1qej2i6m2bv62ox21h1q
+                                                                    var extractedLink=/<a\b.+href="([^\n\r]*)"\s/ig.exec(traceLabel)[1];
+                                                                    var linkWithContextParams =extractedLink+qCharacter+"pcid="+contextUrlParams.cid+"&ppid="+contextUrlParams.pid;
+                                                                    values.legend_markdown_pattern.push(linkWithContextParams);
+                                                                    traceLabel=traceLabel.replace(extractedLink,linkWithContextParams)
+                                                                }catch(error){
+                                                                    traceLabel=row[trace.legend_col];
+                                                                }
+                                                                //delete $rootScope.templateParams;
+                                                            }
+                                                            values.labels.push(traceLabel);
+                                                            //Added the legend column names to text variable so that the tooltip name does not contain a link if the legends contains a link
+                                                            var tooltip=trace.hovertemplate_display_pattern ? ERMrest.renderHandlebarsTemplate(trace.hovertemplate_display_pattern, $rootScope.templateParams) : row[trace.legend_col];
+                                                            values.text.push(tooltip);
                                                         }
                                                         values.values.push(formatData(row[trace.data_col], plot.config ? plot.config.format_data : false, "pie"));
+                                                        if(trace.hasOwnProperty("graphic_link_pattern")){
+                                                            var qCharacter = trace.graphic_link_pattern.indexOf("?") !== -1 ? "&" : "?";
```

Comment:
The "qCharacter" is being extracted in multiple places, can you turn this into a function that is being called instead?

Part of the issue I'm seeing is it is done one way here and another way above on line 706.